src/controller/CHIPDeviceController.cpp

```diff
@@ -2921,7 +2952,28 @@ void DeviceCommissioner::PerformCommissioningStep(DeviceProxy * proxy, Commissio
         NetworkCommissioning::Commands::ConnectNetwork::Type request;
         request.networkID = params.GetWiFiCredentials().Value().ssid;
         request.breadcrumb.Emplace(breadcrumb);
-        CHIP_ERROR err = SendCommand(proxy, request, OnConnectNetworkResponse, OnBasicFailure, endpoint, timeout);
+
+        CHIP_ERROR err = CHIP_NO_ERROR;
+        GeneralCommissioning::Attributes::SupportsConcurrentConnection::TypeInfo::Type supportsConcurrentConnection;
+        supportsConcurrentConnection = params.GetSupportsConcurrentConnection().Value();
+        ChipLogProgress(Controller, "SendCommand kWiFiNetworkEnable, supportsConcurrentConnection=%d",
+                        supportsConcurrentConnection);
+        if (supportsConcurrentConnection)
+        {
+            err = SendCommand(proxy, request, OnConnectNetworkResponse, OnBasicFailure, endpoint, timeout);
+        }
+        else
+        {
+            // Concurrent Connections not allowed. Send the ConnectNetwork command but do not wait for the
+            // ConnectNetworkResponse on the Commissioing network as it will not be present. Stop the timeout
+            // and run what would have been in the onConnectNetworkResponse callback.
+            SendCommand(proxy, request, NonConcurrentNetworkResponse, NonConcurrentTimeout, endpoint, NullOptional);
+            // As there will be no ConnectNetworkResponse, it is an implicit kSuccess so a default report is fine
+            CommissioningDelegate::CommissioningReport report;
+            DeviceCommissioner * commissioner = static_cast<DeviceCommissioner *>(this);
+            commissioner->CommissioningStageComplete(CHIP_NO_ERROR, report);
```

Comment:
This seems like a complicated way of writing:
```suggestion
            CommissioningStageComplete(CHIP_NO_ERROR, report);
```
But also: after you call `CommissioningStageComplete` you must not run any more code in this function...  Needs to return after that call.